server/src/main/java/org/elasticsearch/cluster/routing/allocation/RoutingAllocation.java

```diff
@@ -73,6 +72,11 @@ public class RoutingAllocation {
 
     private final Map<String, SingleNodeShutdownMetadata> nodeReplacementTargets;
 
+    private final Map<String, SingleNodeShutdownMetadata> nodeShutdowns;
+
+    @Nullable
+    private final DesiredNodes desiredNodes;
```

Comment:
Do we need above to make it inline?

Reply:
This isn't really about inlining. This is just faster outright because we don't have to lookup from the customs map for every shard.